### DISCUSSION TITLE
What exactly is the problem with __iter__?

### DISCUSSION THREAD

**Discussion by MonsterAzi** [gabrieldemarmiesse/mojo-stdlib-extensions#39] [General]
In the readme, `__iter__`  is stated to be missing from Mojo, but Mojo definitely has an `__iter__`  method. Is this an oversight or is the readme just poorly worded?

**Comment by gabrieldemarmiesse** [comment]
It's impossible to use it with structs :(

**Reply by MonsterAzi** [reply]
That's not true?? I've used it before and it worked fine?

**Reply by MonsterAzi** [reply]
To be fair, it's really weird the way you have to do it.

**Reply by MonsterAzi** [reply]
I had to implement it when I made a cube struct to iterate through the faces, do you perhaps mean using it within a struct, cause I haven't tried that?

**Reply by guidorice** [reply]
I wonder if this is basically the same issue with __iter__ as here: https://github.com/modularml/mojo/issues/1188
So in other words, it depends if your type is register-passable or not.
Lots of stuff is blocked by the traits feature. I think the mojo iterator support should improve massively after traits.
